runtime/compiler/x/i386/codegen/IA32JNILinkage.cpp

```diff
@@ -338,8 +338,19 @@ TR::Register *TR::IA32JNILinkage::buildJNIDispatch(TR::Node *callNode)
    switch (resolvedMethod->returnType())
       {
       case TR::Int8:
-         generateRegRegInstruction(isUnsigned ? MOVZXReg4Reg1 : MOVSXReg4Reg1,
-                                   callNode, ecxReal, eaxReal, cg());
+         if (comp()->getSymRefTab()->isReturnTypeBool(callSymRef))
+            {
+            // For bool return type, must check whether value returned by
+            // JNI is zero (false) or non-zero (true) to yield Java result
+            generateRegRegInstruction(TEST1RegReg, callNode, eaxReal, eaxReal, cg());
+            generateRegInstruction(SETNE1Reg, callNode, ecxReal, cg());
+            generateRegRegInstruction(MOVZXReg4Reg1, callNode, ecxReal, ecxReal, cg());
```

Comment:
This line is nearly identical (minus the `isUnsigned` check) with the `else` case, maybe we can common them into a fall-through. I'm not sure the value of `isUnsigned` here; however, `ecxReal` can only be 0 or 1 in this case, zero-extension and sign-extension yield same result.

Reply:
I thought about doing that, but then I decided to put the result of `SETNE` into `ecxReal` instead of overwriting `eaxReal`.  I'll switch back to use your suggestion.

Reply:
Revised in commit https://github.com/eclipse/openj9/pull/4086/commits/fefc9a56c812601030c115d541c1a6c8f606f6bb